artsy/force
# [Toolchain] Precompile server-side code with Webpack

Fixes https://artsyproduct.atlassian.net/browse/PLATFORM-1213

This PR removes JIT (just in time) compilation from Force. Previously, on production, we would install `@babel/register` and `coffeescript/register` in order to compile our server-side source-code on the fly, which impacts performance by a) requiring compilation during the request cycle; and b) not optimizing JS for production, which can greatly increase runtime speed. 

The new flow is as follows: 
1. On deploy, `yarn build:server` is run in the Dockerfile
1. This executes a new new Webpack `serverConfig` file
1. A minified `server.dist.js` file is compiled into the root of the app
1. The  server starts by booting  `node server.dist.js`.

**To test**:
```
yarn assets
yarn build:server
yarn start:prod
```

Additional changes:
- Deleted some old example / prototype apps (react, isomorphic relay)
- Converted our new webpack setup to CommonJS from ES6. Throughout this process I noticed numerous issues involving the boundary between compiled code and native node code and by keeping this layer as close to the metal as it can be it removes future ambiguity (until ES modules are supported unconditionally). Note that Webpack supports a `webpack.config.babel.js` setup but this doesn't read in `.babelrc` and is in general a bit buggy.
